### DISCUSSION TITLE
Specific starting position & size for caption

### DISCUSSION THREAD

**Discussion by crcucb** [ImageMagick/ImageMagick#4070] [Help]
I am personalizing a generic office image background for Zoom & Teams meeting.  I added my name & info, logo, picture of the wife & I, and I use convert to add the date to a specific spot.  I run a batch file when I login which calls converter.exe
In the image is a whiteboard.  I want to add variable length/wrapped on the whiteboard as well.  I was first using
I am using this method to add the date:
convert.exe infile.jpg -font Times-New-Roman-Bold-Italic -pointsize 32 -fill navy -draw "text 1200,280 '%curdate%'"  outfile.jpg
I did not find a way to use the -draw text with word wrapping, but I saw references to adding captions.

convert.exe testin.jpg -font Times-New-Roman-Bold-Italic -pointsize 32 -size 165x200 -fill navy -background none caption:"%outtext%" -composite  testout0.jpg
However, I am not sure how to position this.  I was experimenting with the -Gravity which can place it in an area but I want to specific a specific x,y point for the are that I want the text to be contained in.
I also experimented with -region with -gravity but wasn’t able to figure it out.

**Comment by fmw42** [comment]
Include the option -geometry +X+Y which will be relative to whatever -gravity you use.

See https://legacy.imagemagick.org/Usage/layers/#convert

**Comment by crcucb** [comment]
This is exactly what I was looking for.  Thank you.

Second question, is there a way to autosize the font (pointsize) so it fits inside the  area space (-size 385x230) ?

convert.exe" "testin.jpg" -font Times-New-Roman-Bold-Italic -pointsize 32 -size 385x230 -fill navy -gravity Center  -background white caption:"%outtext%" -geometry +280-100 -composite  "C:\ProgramData\Background Image\testout2b.jpg" 

If the entire text doesn't fit inside of the 385x230 space, automatically make the pointsize smaller?

**Comment by snibgo** [comment]
> If the entire text doesn't fit inside of the 385x230 space, automatically make the pointsize smaller?

Not exactly. If you remove the "-pointsize 32", IM will choose the largest pointsize that fits in the specified "-size".
